utilities/pipelines/resourceDeployment/Get-TemplateDeploymenWhatIf.ps1

```diff
@@ -0,0 +1,185 @@
+Optional. Additional parameters you can provide with the deployment. E.g. @{ resourceGroupName = 'myResourceGroup' }
+
+.EXAMPLE
+Get-TemplateDeploymenWhatIf -templateFilePath 'C:/key-vault/vault/main.bicep' -parameterFilePath 'C:/key-vault/vault/.test/parameters.json' -location 'WestEurope' -resourceGroupName 'aLegendaryRg'
+
+Get What-If deployment result for the main.bicep of the KeyVault module with the parameter file 'parameters.json' using the resource group 'aLegendaryRg' in location 'WestEurope'
+
+.EXAMPLE
+Get-TemplateDeploymenWhatIf -templateFilePath 'C:/key-vault/vault/main.bicep' -location 'WestEurope' -resourceGroupName 'aLegendaryRg'
+
+Get What-If deployment result for the main.bicep of the KeyVault module using the resource group 'aLegendaryRg' in location 'WestEurope'
+
+.EXAMPLE
+Get-TemplateDeploymenWhatIf -templateFilePath 'C:/resources/resource-group/main.json' -parameterFilePath 'C:/resources/resource-group/.test/parameters.json' -location 'WestEurope'
+
+Get What-If deployment result for the main.json of the ResourceGroup module with the parameter file 'parameters.json' in location 'WestEurope'
+#>
+function Get-TemplateDeploymenWhatIf {
+
+    [CmdletBinding(SupportsShouldProcess)]
+    param (
+        [Parameter(Mandatory)]
+        [string] $templateFilePath,
+
+        [Parameter(Mandatory)]
+        [string] $location,
+
+        [Parameter(Mandatory = $false)]
+        [string] $parameterFilePath,
+
+        [Parameter(Mandatory = $false)]
+        [string] $resourceGroupName,
+
+        [Parameter(Mandatory = $false)]
+        [string] $subscriptionId,
+
+        [Parameter(Mandatory = $false)]
+        [string] $managementGroupId,
+
+        [Parameter(Mandatory = $false)]
+        [Hashtable] $additionalParameters
+    )
+
+    begin {
+        Write-Debug ('{0} entered' -f $MyInvocation.MyCommand)
+
+        # Load helper
+        . (Join-Path (Get-Item -Path $PSScriptRoot).parent.FullName 'sharedScripts' 'Get-ScopeOfTemplateFile.ps1')
+    }
+
+    process {
+        $DeploymentInputs = @{
+            TemplateFile = $templateFilePath
+            Verbose      = $true
+            OutVariable  = 'ValidationErrors'
+        }
+        if (-not [String]::IsNullOrEmpty($parameterFilePath)) {
+            $DeploymentInputs['TemplateParameterFile'] = $parameterFilePath
+        }
+        $ValidationErrors = $null
+
+        # Additional parameter object provided yes/no
+        if ($additionalParameters) {
+            $DeploymentInputs += $additionalParameters
+        }
+
+        $deploymentScope = Get-ScopeOfTemplateFile -TemplateFilePath $templateFilePath -Verbose
+
+        $deploymentNamePrefix = Split-Path -Path (Split-Path $templateFilePath -Parent) -LeafBase
+        if ([String]::IsNullOrEmpty($deploymentNamePrefix)) {
+            $deploymentNamePrefix = 'templateDeployment-{0}' -f (Split-Path $templateFilePath -LeafBase)
+        }
+        if ($templateFilePath -match '.*(\\|\/)Microsoft.+') {
```

Comment:
Question @tyconsulting , I assume your module structure still has the `Microsoft.` prefix for the ProviderNamespace folder? Upstream we removed that prefix in summer as a first step towards the Public Bicep Registry. In turn we may need to adjust this code accordingly and match/split at `modules`.